holdings_maintenance/pds4/pds4indexshelf.py

```
@@ -0,0 +1,541 @@
+                        dt.strftime('%Y-%m-%dT%H-%M-%S'), force=True)
+
+        else:
+            logger.info('!!! Index shelf file is up to date; repair canceled',
+                        shelf_path, force=True)
+
+        return
+
+    # Write new info
+    write_indexdict(pdsf, table_indexdict, logger=logger)
+
+def update(pdsf, selection=None, logger=None):
```

Ruff (0.15.2)

[warning] Unused function argument: `selection`

(ARG001)